source/appModules/kindle.py

```diff
@@ -0,0 +1,139 @@
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2016 NV Access Limited
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+import time
+import appModuleHandler
+import speech
+import sayAllHandler
+import eventHandler
+import api
+from scriptHandler import willSayAllResume, isScriptWaiting
+import controlTypes
+import treeInterceptorHandler
+import IAccessibleHandler
+from cursorManager import ReviewCursorManager
+from browseMode import BrowseModeDocumentTreeInterceptor
+import textInfos
+from textInfos import DocumentWithPageTurns
+from NVDAObjects.IAccessible import IAccessible, IA2TextTextInfo, getNVDAObjectFromEvent
+
+class BookPageViewTreeInterceptor(DocumentWithPageTurns,ReviewCursorManager,BrowseModeDocumentTreeInterceptor):
+
+	TextInfo=treeInterceptorHandler.RootProxyTextInfo
+
+	def turnPage(self,previous=False):
+		try:
+			self.rootNVDAObject.appModule.inPageTurn=True

```

Comment:
Why is this needed? I think we need a comment here, possibly referring to the focus ignorer thingy.